### PR TITLE
[datadog_synthetics_test] Allow deletion of Synthetics test steps

### PR DESCRIPTION
**Pull request by ALenfant**
Following https://github.com/DataDog/dogweb/pull/150819, our Synthetics test edit API endpoints now take into account empty arrays.

This PR:
* Always sends the step arrays for Browser and Mobile tests, meaning if no steps are defined, an empty array is sent and any existing steps will be removed on apply
* Adds tests for this edge case
* Updates an existing test where a subtest had no steps and thus now sends an empty steps array

https://datadoghq.atlassian.net/browse/SYNTH-22099

### CODE REVIEW

**Review by HantingZhang2 — APPROVED** [review]

File: datadog/tests/cassettes/TestAccDatadogSyntheticsBrowserTestBrowserNewBrowserStep_Basic.yaml [hunk 1]
```diff
@@ -67,13 +67,13 @@ interactions:
...
+            {"public_id":"fty-6cs-gk4","name":"tf-TestAccDatadogSyntheticsBrowserTestBrowserNewBrowserStep_Basic-local-1760430810-subtest","status":"paused","type":"browser","tags":[],"created_at":"2025-10-14T08:33:34.146847+00:00","modified_at":"2025-10-14T08:33:34.146847+00:00","config":{"assertions":[],"configVariables":[],"request":{"method":"GET","url":"https://www.datadoghq.com"},"variables":[]},"message":"","options":{"device_ids":["laptop_large"],"httpVersion":"any","min_location_failed":1,"monitor_options":{"escalation_message":"","renotify_interval":120,"renotify_occurrences":0,"on_missing_data":"show_no_data","notify_audit":false,"new_host_delay":300,"include_tags":true},"retry":{"count":2,"interval":300},"tick_every":900},"locations":["aws:eu-central-1"],"monitor_id":226800901,"creator":{"name":"Antonin Lenfant Kodia","handle":"antonin.lenfant@datadoghq.com","email":"antonin.lenfant@datadoghq.com"},"steps":[]}
```
Comment (HantingZhang2) [inline datadog/tests/cassettes/TestAccDatadogSyntheticsBrowserTestBrowserNewBrowserStep_Basic.yaml]: Nit: this is a public repo and you are showing your name and work email in the cassette, use shared credential in the frog org if you want to avoid that
